bothan-lib/src/metrics/rest.rs

```diff
@@ -0,0 +1,86 @@
+use opentelemetry::metrics::{Counter, Histogram};
+use opentelemetry::{KeyValue, global};
+
+pub enum RequestStatus {
+    Success,
+    Failed,
+    Timeout,
+}
+
+impl RequestStatus {
+    fn as_str_name(&self) -> &'static str {
+        match self {
+            RequestStatus::Success => "success",
+            RequestStatus::Failed => "failed",
+            RequestStatus::Timeout => "timeout",
+        }
+    }
+}
+
+pub enum ResponseLatencyStatus {
+    Success,
+    Failed,
+}
+
+impl ResponseLatencyStatus {
+    fn as_str_name(&self) -> &'static str {
+        match self {
+            ResponseLatencyStatus::Success => "success",
+            ResponseLatencyStatus::Failed => "failed",
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct RestMetrics {
+    total_requests: Counter<u64>,
+    response_latency: Histogram<u64>,
+}
+
+impl Default for RestMetrics {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl RestMetrics {
+    pub fn new() -> Self {
+        let meter = global::meter("rest_source");
```

Comment:
shouldnt this be specific to each source? you would need the source name as the input here